custom_components/tahoma/cover.py

```diff
@@ -169,14 +172,19 @@ def current_cover_tilt_position(self):
         None is unknown, 0 is closed, 100 is fully open.
         """
         position = self.select_state(CORE_SLATS_ORIENTATION_STATE)
-        return 100 - position if position is not None else None
+        if position is None:
+            return None
+        elif self.device.widget == "PositionableExteriorVenetianBlind":
+            return position
+        else:
+            return 100 - position
 
     async def async_set_cover_position(self, **kwargs):
         """Move the cover to a specific position."""
         position = 100 - kwargs.get(ATTR_POSITION, 0)
 
         # HorizontalAwning devices need a reversed position that can not be obtained via the API
-        if "Horizontal" in self.device.widget:
+        if "Horizontal" in self.device.widget or "AwningValance" == self.device.widget:
```

Comment:
By the way, all of those are Awnings. What if we just say:
`if self.device.uiclass == "Awning` 

It looks like they made all the Awnings reversed since the last update, which means we can maybe remove the ugly "Horizontal" hack?

Reply:
agreed it's better